com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/Project.java

```diff
@@ -1458,13 +1426,31 @@ private void configurePreBuildProjectOptions() throws IOException, CompileExcept
         options.add(new UniOption("model-split-large-meshes", "model-split-large-meshes", "model","split_meshes",null));
         options.add(new UniOption("prometheus-disabled", "prometheus-disabled", "prometheus","disabled",null));
 
+        Platform currentPlatform = getPlatform();
+        final List<Platform> architectures = Platform.getArchitecturesFromString(this.option("architectures", ""), currentPlatform);
+        Set<String> architectureSet = new HashSet<>();
+        for (Platform platform : architectures) {
+            architectureSet.add(platform.toString());
+        }
+        for (String path : currentPlatform.getExtenderPaths()) {
+            architectureSet.add(path);
+        }
+        List<Map<String, Object>> platformsSettings = new ArrayList<>();
+        for(String arch : architectureSet) {
+            platformsSettings.add(ExtenderUtil.getPlatformSettings(this, arch));
+        }
```

Comment:
We should take into account not only the current platform, but all the architectures + generic extender path (e.g. `android` for `armv7-android`)